app/validators/date-range.js

```diff
@@ -0,0 +1,7 @@
+import moment from 'moment'
+
+export default function validateDateRange(options) {
+  return (key, newValue) => {
+    return moment().range(options.min, options.max).contains(moment(newValue), false)
```

Comment:
Help wanted: I don't know how to access the store here, to be able to get the challenge model for the startsOn value!